internal/resources/gsi.go

```diff
@@ -234,6 +234,20 @@ This will automatically be retried in the background.  Please run "terraform app
 			),
 		)
 
+		return
+	case internalerrors.ErrGatewayTimeoutForIndexDDL:
+		resp.Diagnostics.AddWarning(
+			"Long index build time",
+			fmt.Sprintf(
+				`Build of index %s in %s.%s.%s did not complete in time.  This maybe due to non-deferred builds of a large index,
```

Comment:
is using deferred builds the only solution? what if they have fired the non-deferred index, what can they do next?